### ISSUE TITLE
[BUG] Shortcut for Refresh fails with "the operation took too long to complete"

### ISSUE THREAD

**Issue by hotdog5785** [SideStore/SideStore#431]
### Describe the bug

error message occurs; the operation took too long to complete. please try again

### Instructions to reproduce

ios 16.4 & ios 16.6
in shortcut, only use 'refresh all apps' - of course, with vpn
after error message, I can confirm sidestore reinstallation with blue dot.
maybe refresh itself could be successful... I will experiment this case next day.  (it's my first day using sidestore)

### What version of SideStore are you using?

0.5.0

**Comment by hotdog5785** [comment]
all apps successfully refreshed with error message.

**Comment by naturecodevoid** [comment]
Known issue

**Comment by yur1xpp** [comment]
Do we have any update on this? I see some similar issue here https://github.com/SideStore/SideStore/issues/337 too.

**Comment by naturecodevoid** [comment]
SideStore is mostly run by volunteers in our free time. Additionally, we sometimes prefer to work on other projects in our limited free time. There are also a lot of people on the team who don't have a Mac to work on SideStore. This is why we usually don't have ETAs. We are also usually more focused on larger issues.

**Comment by naturecodevoid** [comment]
You're completely fine! I don't think this has been fixed. The app expire bug was very high priority, so that was focused on more than this one.

**Comment by reyland44** [comment]
Still getting this error unfortunately

**Comment by Raffaeledefilippo** [comment]
Sadly me too, the error happens almost every time, seldom a single app refreshes though and even more rarely, like almost never, all the apps refresh with the operation ending successfully

**Comment by mmupmip** [comment]
Just commenting to get notified when there's an update as this is still happening

**Comment by yodaluca23** [comment]
> Just commenting to get notified when there's an update as this is still happening

Just so you know, you can click subscribe on the right side, you don't have to comment...

![image](https://github.com/user-attachments/assets/231880dd-b3c7-4f4b-8db6-62a2dfe6d805)

**Comment by Infinitay** [comment]
Would a potential solution be to offer Shortcut actions to refresh each app individually? People would have to create a new shortcut action but assuming it works with no issue, it'll work reliably. Further more, they could create it once, duplicate and modify the single app to resign. People could still use refresh all if they have a few apps and don't have any existing issue.

**Comment by AhmedNSidd** [comment]
> Would a potential solution be to offer Shortcut actions to refresh each app individually? People would have to create a new shortcut action but assuming it works with no issue, it'll work reliably. Further more, they could create it once, duplicate and modify the single app to resign. People could still use refresh all if they have a few apps and don't have any existing issue.

This could potentially work...

Warning, I'm a complete newbie to all of this, but here's my understanding of the issue so far:

It seems to me the reason why we are getting the "the operation took too long to complete" error is that the `Shortcuts` app probably has some sort of timeout for SideStore's `Refresh all apps` shortcut action. So when our shortcut is run, and SideStore is attempting to refresh its apps, some of its apps take too long to refresh, like, for example, the `YtLitePlus` app can take something like 2 minutes to finish refreshing, whereas the Shortcuts Action only gives the action 10 seconds to run before giving "the operation took too long to complete" error in shortcuts.

**However, it seems peculiar to me that we are able to get SideStore to refresh consistently every single time even with this timeout in question**. I haven't looked at the code to see how this is ending up being done, but if we can somehow leverage this mechanism of being able to refresh an app (even if we get "the operation took too long to timeout" error), then we could potentially build a shortcut that looks something like this:

1. Run Shortcut Action `Refresh YtLitePlus App on SideStore` (this operation will end up giving the error `the operation took too long to timeout`, but again, if we can somehow leverage the mechanism of the Shortcuts still refreshing the app even with this error, this will still end up refreshing `YtLitePlus`, even if we do get the error)
2. Run Shortcut Action `Refresh SideloadedExampleApp2 App on SideStore`
3. Run Shortcut Action `Refresh SideloadedExampleApp3 App on SideStore`
4. Finally, run Shortcut Action `Refresh SideStore App on SideStore`
 
Although now that I have typed all this out, I just realized this won't work. The reason is that when you get the `the operation took too long to complete` error, it stops the rest of the shortcut from running, so in the example above, the shortcut would stop running after Shortcut #1

That said, bear with me a second, but what if you automated multiple different shortcuts with time intervals between each shortcut? Here's an example:

1. Create a shortcut with a single action: `Refresh YtLitePlus on SideStore`. Run this shortcut every day at 6 am.
2. Create another shortcut with a single action: `Refresh SideloadedExampleApp2 on SideStore`. Run this shortcut every day at 6:05 am.
3. Create another shortcut with a single action: `Refresh SideloadedExampleApp3 on SideStore`. Run this shortcut every day at 6:10 am.
4. Create another shortcut with a single action: `Refresh SideStore App on SideStore`. Run this shortcut every day at 6:15 am.

Even though each of these shortcuts would given an error of `the operation took too long to complete`, it could still theoretically be refreshing the apps in the background, and then at the end we would be refreshing sidestore... I have no idea if this would work..
